Signal/src/AppDelegate.m

```diff
@@ -90,13 +101,15 @@ - (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(
 
     self.notificationTracker = [NotificationTracker notificationTracker];
     
-    [self performUpdateCheck];
     CategorizingLogger* logger = [CategorizingLogger categorizingLogger];
     [logger addLoggingCallback:^(NSString *category, id details, NSUInteger index) {}];
     [Environment setCurrent:[Release releaseEnvironmentWithLogging:logger]];
     [Environment.getCurrent.phoneDirectoryManager startUntilCancelled:nil];
     [Environment.getCurrent.contactsManager doAfterEnvironmentInitSetup];
     
+    [self performUpdateCheck]; // this call must be made after environment has been initialized because in general upgrade may depend on environment
```

Comment:
this was a second place the existing migration had broken, performUpdateCheck must be called after we have an environment